### PR TITLE
Documentation for Pipelines API

### PR DESCRIPTION
**Pull request by mattppal**
# Summary

Add V1 of documentation for our [Pipelines API](https://docs.mage.ai/contributing/backend/api/pipelines). This will resolve https://github.com/mage-ai/mage-ai/issues/2883.

Notes:
- This is quickly turning into a rabbit hole, i.e. documenting schedules and blocks payloads.
- We might need to find a more reusable structure for docs, I'm repeating a lot of info/code here.
  - Specifically, I think components like [these](https://github.com/mage-ai/mage-ai/pull/2907/files#diff-7e66c099b07f5654f7694fd895a71670482dbe2f79aebacd20ea628b1a004f44R397) could be tricky, i.e. parameters with complex payloads.
  - 

# Tests
Tested using Mintlify locally.

### CODE REVIEW

File: docs/contributing/backend/api/pipelines.mdx [hunk 1]
```diff
@@ -241,29 +250,262 @@ curl --request GET \
...
+  headers = {
+    "Content-Type": "application/json",
+    "X-API-KEY": "zkWlN0PkIKSN0C11CfUHUj84OT5XOJ6tDZ6bDRO2",
+    "Cookie": "oauth_token=some_really_long_string"
+  }
+
+  json_data = {
+    'pipeline': {
+            "type": 'integration'
+            }
```
Comment (wangxiaoyou1993) [inline docs/contributing/backend/api/pipelines.mdx:434]: nit: indent

File: docs/contributing/backend/api/pipelines.mdx [hunk 2]
```diff
@@ -241,29 +250,262 @@ curl --request GET \
...
+
+    headers = {
+      "Content-Type": "application/json",
+      "X-API-KEY": "zkWlN0PkIKSN0C11CfUHUj84OT5XOJ6tDZ6bDRO2",
+      "Cookie": "oauth_token=some_really_long_string"
+    }
+
+    json_data = {
+      'pipeline': {
+            "name": "arwen-starlight",
```
Comment (wangxiaoyou1993) [inline docs/contributing/backend/api/pipelines.mdx:355]: nit: 2 spaces indentation

File: docs/contributing/backend/api/pipelines.mdx [hunk 3]
```diff
@@ -241,29 +250,262 @@ curl --request GET \
...
+    "Cookie": "oauth_token=some_really_long_string"
+  }
+  
+  r = requests.get('http://localhost:6789/api/pipelines/arwen_starlight', headers=headers)
+  ```
+</CodeGroup>
+
+```json
+{
+    "pipeline": {
```
Comment (wangxiaoyou1993) [inline docs/contributing/backend/api/pipelines.mdx:283]: nit: 2 spaces indentation

File: docs/contributing/backend/api/pipelines.mdx [hunk 4]
```diff
@@ -241,29 +250,262 @@ curl --request GET \
...
+            }
+  }
+  
+  r = requests.put('http://localhost:6789/api/pipelines/arwen_starlight', headers=headers, json=json_data)
+  ```
+</CodeGroup>
+
+```json
+{
+    "pipeline": {
```
Comment (wangxiaoyou1993) [inline docs/contributing/backend/api/pipelines.mdx:443]: nit: 2 spaces

File: docs/contributing/backend/api/pipelines.mdx [hunk 5]
```diff
@@ -241,29 +250,262 @@ curl --request GET \
...
+    "Cookie": "oauth_token=some_really_long_string"
+  }
+  
+  r = requests.delete('http://localhost:6789/api/pipelines/arwen_starlight', headers=headers)
+  ```
+</CodeGroup>
+
+```json
+{
+    "pipeline": {
```
Comment (wangxiaoyou1993) [inline docs/contributing/backend/api/pipelines.mdx:491]: nit: 2 spaces

File: docs/contributing/backend/api/pipelines.mdx [hunk 6]
```diff
@@ -6,48 +6,47 @@ block dependencies."
 
 ## Pipeline object
 
-
 ```json
 {
   "blocks": [
-    {
```
Comment (wangxiaoyou1993) [inline docs/contributing/backend/api/pipelines.mdx:13]: You don't need to remove the indent here

**Comment by wangxiaoyou1993** [comment]
> Do we have any way to lint these, currently?

I don't have linter for the doc file. For the json blob, i think you can format it separately and copy it here.

**Review by wangxiaoyou1993 — APPROVED** [review]

File: docs/contributing/backend/api/pipelines.mdx (hunk 6 above)
Comment (mattppal) [reply docs/contributing/backend/api/pipelines.mdx:13]: I used VS code to convert from four spaces to two and this was a byproduct. I'll fix

**Pull request merged**